node/resolvers/Routes/index.ts

```diff
@@ -168,6 +259,36 @@ const Index = {
 
     ctx.response.status = 200
   },
+  requestCancellation: async (ctx: Context) => {
+    const {
+      vtex: {
+        route: {
+          params: { orderId },
+        },
+      },
+      clients: { checkout },
+    } = ctx
+
+    if (!orderId) {
+      throw new UserInputError('Order ID is required')
+    }
+
+    // use order route handler to verify user has access to order
+    Index.order(ctx).catch(() => {
+      throw new ForbiddenError('Access denied')
+    })
```

Comment:
Is it necessary to call the Index.order and pass through to the entire route only to check the permission? I'd create a new function to check it and call it in both routes. Even though, this throw could be called after the response calls once it's not into a try/catch + async/await statement. What do you think?

Reply:
Sure, it's worth a shot 👍

Reply:
Done